src/test/ui/impl-trait/impl-generic-mismatch-ab.stderr

```diff
@@ -5,7 +5,11 @@ LL |     fn foo<A: Debug>(&self, a: &A, b: &impl Debug);
    |                                -- type in trait
 ...
 LL |     fn foo<B: Debug>(&self, a: &impl Debug, b: &B) { }
-   |                                ^^^^^^^^^^^ expected type parameter, found a different type parameter
+   |            -                   ^----------
+   |            |                   ||
+   |            |                   |found type parameter
+   |            |                   expected type parameter `B`, found type parameter `impl Debug`
+   |            expected type parameter
```

Comment:
It would be nice to do a `span.contains()` check so that we emit this instead:

```
LL |     fn foo<B: Debug>(&self, a: &impl Debug, b: &B) { }
   |            -                   ^^^^^^^^^^^ expected type parameter `B`, found type parameter `impl Debug`
   |            |                   
   |            expected type parameter
```